libs/ml/include/ml/ops/ops.hpp

```diff
@@ -95,7 +95,9 @@ class Trainable
   virtual struct fetch::ml::StateDict<T> StateDict() const                            = 0;
   virtual void              LoadStateDict(struct fetch::ml::StateDict<T> const &dict) = 0;
   virtual TensorType const &get_weights() const                                       = 0;
-  virtual TensorType const &get_gradients() const                                     = 0;
+  virtual void              set_weights(TensorType &new_value)                        = 0;
```

Comment:
weights and gradients are NOT common to all ops. Only to trainable ops. Presumably these virtual methods should exist in trainable and not Ops?

Also, by setting this pure virtuals here it presumably breaks every Op for which you haven't implemented these methods? (because they're now abstract classes)